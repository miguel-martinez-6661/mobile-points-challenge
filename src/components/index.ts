export * from './text/text';
export * from './button/button';
export * from './card/card';
export * from './loading-indicator/loading-indicator';
