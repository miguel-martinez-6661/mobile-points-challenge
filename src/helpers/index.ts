export * from './numbers-helpers';
export * from './date-helpers';
