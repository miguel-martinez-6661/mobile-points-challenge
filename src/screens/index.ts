export * from './home/home-screen';
export * from './details/product-details-screen';
