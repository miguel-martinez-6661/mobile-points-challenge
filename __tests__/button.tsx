import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Button } from '../src/components';

describe('Testing <Button> component', () => {
  it('should mount component correctly', () => {
    const buttonComponent = renderer
      .create(<Button title="Test Button" />)
      .toJSON();
    expect(buttonComponent).toMatchSnapshot();
  });

  it('should mount component correctly and support an onPress event', () => {
    const onPressEvent = jest.fn();
    const buttonComponent = renderer.create(
      <Button title="Test Button" onPress={onPressEvent} />,
    );
    buttonComponent.root.findByType(Button).props.onPress();
    expect(onPressEvent).toHaveBeenCalled();
  });
});
