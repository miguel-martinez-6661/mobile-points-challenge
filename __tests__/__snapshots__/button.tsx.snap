// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Button> component should mount component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "display": "flex",
      "height": 50,
      "opacity": 1,
      "padding": 14,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Avenir",
        "fontSize": 22,
        "fontWeight": "900",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
